hypothesis-python/src/hypothesis/_settings.py

```diff
@@ -748,7 +757,24 @@ def note_deprecation(
     warnings.warn(HypothesisDeprecationWarning(message), stacklevel=2 + stacklevel)
 
 
-settings.register_profile("default", settings())
+DEVELOPMENT = settings()
+
+CI = settings(
+    derandomize=True,
+    deadline=None,
+    database=None,
+    print_blob=True,
+    suppress_health_check=list(HealthCheck),
+)
+
+settings.register_profile("ci", CI)
+settings.register_profile("development", DEVELOPMENT)
+
+
+# This is tested in a subprocess so the branch doesn't show up in coverage.
+settings.register_profile(
+    "default", CI if is_in_ci() else DEVELOPMENT
+)  # pragma: no branch
```

Comment:
- I really don't want the content of any given profile to vary based on the environment.
  - How about we also register a `default_ci` profile, and activate that if the CI env var is set?
- I'd keep the database enabled, to support e.g. [`GitHubArtifactDatabase`](https://hypothesis.readthedocs.io/en/latest/database.html#hypothesis.database.GitHubArtifactDatabase)
- We have special handling for `print_blob`-in-CI somewhere which should be undone as redundant with this change.